### DISCUSSION TITLE
Menu new design

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#82] [Q&A] [answered]
We can redesign the app menu with large icon buttons for better usability.

(Android deprecated the drawer, e.g. Google Play and Google Maps apps)

Similar to:
- https://github.com/devemux86/cruiser/discussions/57
- https://github.com/devemux86/cruiser/discussions/58

<img src="https://user-images.githubusercontent.com/3484020/225854856-de63a396-fa89-48d8-977c-5afa7231f815.jpg" width="33%"/><img src="https://user-images.githubusercontent.com/3484020/225854868-7caa9bd4-5755-4221-9f38-dc0f9d85403a.jpg" width="33%"/><img src="https://user-images.githubusercontent.com/3484020/225854877-a5affc3f-49f3-4855-b815-2373b21394c0.jpg" width="33%"/>
<img src="https://user-images.githubusercontent.com/3484020/225854890-773d7af5-07af-457b-945b-44647449d64d.jpg" width="33%"/><img src="https://user-images.githubusercontent.com/3484020/225854896-4ef514a9-3da4-4be4-bf53-3006b21dd64b.jpg" width="33%"/>

**Comment by devemux86** [comment]
Preview images as work in progress of an alternative menu in the 1st post (portrait / landscape):
- https://github.com/devemux86/cruiser/discussions/82#discussion-4955158

**Comment by DolomitesTom** [comment]
Like this a lot🤩

**Comment by MotoUKRider** [comment]
A very nice look - and it'll help the velcro on my gloves last longer :)

**Comment by devemux86** [comment]
Preview images of all new menus in the 1st post:
- https://github.com/devemux86/cruiser/discussions/82#discussion-4955158

---

Currently, menu levels 1 and 2 have this new design:
- Main menu
- Menu: Map, Routing, Favorites, Tools

It's easy to convert any selection dialog to the new look,
just tell me which ones you want. :slightly_smiling_face:

**Reply by MotoUKRider** [reply]
Again - looking very nice.  I am wondering whether a "title text" at the top of the popup dialog is needed to tell the user where they are.  Can I flick/swipe the dialog away with my finger or would the phone's "back" button be required?

Example (I also changed the Favorites text to "List"):
![menu-example](https://user-images.githubusercontent.com/125696109/225747493-a39ab1e4-763b-43d1-a226-dd438b392312.jpg)

**Reply by devemux86** [reply]
I thought of adding titles, I started with a simple appearance, expecting your ideas.
Now they can be closed with Android back button or pressing out of the menu area.

Currently when you close a level 2 menu (e.g. Favorites), you return to the app / map screen.
I could make them return to the main menu, but then we would need 2 steps to close them?

**Reply by MotoUKRider** [reply]
Hmmm, tricky!  I'm also thinking about how this might work once the new "offline" search is live, especially if you're "Navigating" and are adding a Via Point whilst sitting in a layby - you may want a quick button/shortcut for "Back to Map/Route" without all the back/back/back/back faffery of the other app ;)

**Reply by devemux86** [reply]
> I also changed the Favorites text to "List"

We will have more favorite categories in the future:
- Favorite points
- https://github.com/devemux86/cruiser/discussions/35
- https://github.com/devemux86/cruiser/discussions/81

**Comment by WolfgangZ42** [comment]
Prefering the title text - just to know where I am. 
Android back button or/and tap outside the menu is perfect. Also returning to map directly, just a tap more for users with more than one change. 

Not everything has to be converted into icon menus I guess - for example this is already readable and managable enough:
![2023-02-19 14 12 55](https://user-images.githubusercontent.com/121806238/225761811-cadeed70-a4e8-4cd6-88bd-40b15c7d3b15.png)

**Reply by devemux86** [reply]
Not everything, at least functions that could be called outside with gloves.

The thing about texts is that they often don't fit the available space.
e.g. large translations, large font size.

For example in the image above, the "Downloads" in 2 lines is not very pretty?
Like browser toolbar, we get used to the icons and learn to use them intuitively?

<img src="https://user-images.githubusercontent.com/3484020/225854584-65494414-ae60-44de-a8a2-9f83108fd305.jpg" width="50%"/>

**Comment by devemux86** [comment]
New preview images of the menus (with titles) in the 1st post:

- https://github.com/devemux86/cruiser/discussions/82#discussion-4955158

**Comment by DolomitesTom** [comment]
I am thinking of the menu entry to skip several waypoints at once. At the moment a list of shaping and waypoints is shown, you have to scroll through the list in potentially small letters, select one and then press ok. Maybe it would be better for overview if all named waypoints (not shaping points) would be listed as own button in a next button list to be able to skip easily with 2 clicks to this interim destination.

**Reply by devemux86** [reply]
> At the moment a list of shaping and waypoints is shown, you have to scroll through the list in potentially small letters, select one and then press ok.

Currently this is a native Android dialog.
They can only be adjusted by the settings "Font size" and "Display size".

> Maybe it would be better for overview if all named waypoints (not shaping points) would be listed as own button in a next button list to be able to skip easily with 2 clicks to this interim destination.

Do you mean when press the "Skip ?" to show a new screen with the next waypoints as large buttons?
By the way, this list should include both via + shaping points, as skip next waypoint(s) uses both types?

Similarly, for navigation action "Avoid roadblock", the screen to select distances, does it need buttons?

**Reply by DolomitesTom** [reply]
> Do you mean when press the "Skip ?" to show a new screen with the next waypoints as large buttons?

yes, this is what I've meant.
Why also showing SPs? Because of the missing curvy routing option, you normally need to set more shaping points in order to use the preferred streets. Therefore shaping points have in princple no meaning. During navigation you can't see how many SPs have to be deleted in order to use a different road. But WPs you normally have in mind. Therefore I find it better to only show WPs as extra buttons for skipping all SPs between GPS position and that WP.

> Similarly, for navigation action 
"Avoid roadblock", the screen to select distances, does it need buttons?

yes, would be much better and then auto-ok when selecting one option.

**Comment by jclaudej** [comment]
Hi

When this "update" will be available (Android 10) ?

**Reply by devemux86** [reply]
Hi @jclaudej!

I am currently working on several related features:
- https://github.com/devemux86/cruiser/discussions/74
- https://github.com/devemux86/cruiser/discussions/45
- https://github.com/devemux86/cruiser/discussions/35
- https://github.com/devemux86/cruiser/discussions/81

This means many new functions, so the menu has also been redesigned.

It's a lot of work with many steps, I could probably do an intermediate release once things settle down a bit.

**Reply by WolfgangZ42** [reply]
Take your time, Emux!
Everything is fine already - today I rode my first biketour of 2023 ... Cruiser was a perfect guide!

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the kind words!

Please let me know how you find the new speed auto zoom in its topic:
- https://github.com/devemux86/cruiser/discussions/63

**Reply by WolfgangZ42** [reply]
As I said in #63 already: everything is fine. 
But: we should talk about the voice announcements please ... 😃

**Reply by devemux86** [reply]
> we should talk about the voice announcements please

Don't worry, I have already put it in my list to check again what can be done: :slightly_smiling_face: 

- https://github.com/devemux86/cruiser/discussions/63#discussioncomment-5293158

**Comment by DolomitesTom** [comment]
A new interim release with menu redesign would be great.

**Comment by devemux86** [comment]
Also, you don't have to be limited to the color of the menus / buttons,
you will be able to customize the appearance of the app as you like:

<img src="https://user-images.githubusercontent.com/3484020/226115652-d72e24b4-568f-4b64-88d0-c76c98f0895a.jpg" width="25%"/><img src="https://user-images.githubusercontent.com/3484020/226113837-8aa31162-b97c-43b2-bdaa-9c63dcfbadd2.jpg" width="25%"/><img src="https://user-images.githubusercontent.com/3484020/226113844-821b76ad-a5e4-46c4-89f5-412af7d2bfed.jpg" width="25%"/><img src="https://user-images.githubusercontent.com/3484020/226113846-19e3ad65-6d99-445e-a49f-0a5e1842e0ff.jpg" width="25%"/>

**Comment by devemux86** [comment]
"Avoid roadblock" large icon buttons will be available in the next Cruiser update:

<img src="https://user-images.githubusercontent.com/3484020/226390314-b43e8550-ee7a-4680-8991-88430571f10a.jpg"/>

**Reply by devemux86** [reply]
> "Avoid roadblock" large icon buttons will be available in the next Cruiser update

Available in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by DolomitesTom** [comment]
new version 3.0.13 looks magnificent

**Reply by devemux86** [reply]
You are very fast! :+1:

**Comment by devemux86** [comment, marked as answer]
New application menu (see [announcement](https://github.com/devemux86/cruiser/discussions/82#discussion-4955158)) is available in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
The color can be customized in "Settings | Application | Appearance | Menu color" or search settings for "color".

Please let me know how you see it.

**Comment by Leeb64** [comment]
Definitely easier once you get used to it.
But i miss the button "Open Map Theme"?  was that on purpose?
...if necessary, you can add a new map style to :gr.talent.cruiser>files>maps>themes...

**Reply by devemux86** [reply]
Indeed, it needs some time to get used to it. :slightly_smiling_face: 
We can improve the user experience, if you need anything please let me know!

> But i miss the button "Open Map Theme"?

The menu buttons "Open maps" and "Open map theme" are available after enable the
"Settings | Application | Advanced | Menu | Advanced" or search settings for "menu".

**Comment by WolfgangZ42** [comment]
Although I am actually more of a text-oriented user: **Congratulations on the redesign of the menu!** Easier and faster operation (even without gloves 😄) and a very attractive design.

**Comment by DolomitesTom** [comment]
Concerning the overview button which now shows the complete route: this now switches between full overview and actual view. So far so fine, but the compass lost his function. Is there a plan to give it a new function. For example the proposed "partly overview" which could switch between different zoom modes (% of the complete route) starting from the actual gps position on every tap, for example 
1st tap: next 20% (with fallback to actual view after timer countdown)
2nd tap (within the timer): next 40% (with fallback to actual view after timer countdown)
3rd tap (within the timer): next 60% (with fallback to actual view after timer countdown)
3th tap (within the timer): 100% (with fallback to actual view after timer countdown)

Since the big buttons are so convenient, it could maybe the better function for the overview button and think of something else for the compass tap.

**Reply by devemux86** [reply]
Thanks for the interesting suggestion!
Compass tap is indeed free now, so we can think of an action to put.

Compass _long press_ still exists to select the map orientation (rotation 2D / 3D or not) in navigation / follow.

@DolomitesTom can you please create a new topic to discuss it there?

**Comment by devemux86** [comment]
If you think there are app dialogs / screens that could be better used if converted with the new menu design with large icon buttons,
please let me know!

**Reply by MotoUKRider** [reply]
The new/larger icons UI is very good and does make some actions easier.

I've mentioned this one previously, but the Waypoints list isn't great in the landscape orientation. The list isn't big enough to scroll and I think there's wasted screen real-estate.  Perhaps buttons on the left/right so the list can use the full "height" of the window when in landscape?  I'm guessing at least 3 list items visible would be ideal.

Maybe remove the Shaping/Via-Point button and have that as the list item's icon which you can touch to toggle?

Single-line distance/time as there's lots of room along the single line :)

What I see on my Samsung S20:
![waypoints-landscape](https://user-images.githubusercontent.com/125696109/229625680-2dfad326-8b36-4ca6-b3e2-15bfce4de146.jpg)

I should point out that I'm using a new phone mount (sprung-loaded holder) which sits behind my screen BMW Navigator style, via an AMPS plate - so I can't rotate it without taking it out of the cradle to hold it :)  My thinking will most-likely be sideways from now on!

**Reply by DolomitesTom** [reply]
Was there 3.0.16 ? I still don't see it. I also used to use the portrait mode, but since 1 BMW within the bikes is limiting this, I have also switched to the landscape mode since a while 🙈.

**Reply by DolomitesTom** [reply]
Now I know what you mean. I am using that menu not too often. The headline and cancel and ok button is too big with big font setting. Maybe force the headline and bottom line with small font independent of font settings. Or leave the headline completly out, I don't need it, and make the cancel and ok Button as Icons in the button line?

**Reply by DolomitesTom** [reply]
Speaking of which, is drag and drop possible in this menu to be able to move points? The competition has that too 😁. If not, no problem, not a big issue🙂.  
On the other hand, in landscape mode with a large font and few elements, this is probably rather counterproductive.

**Reply by devemux86** [reply]
@MotoUKRider @DolomitesTom thanks for the suggestions!

I created a new topic with your ideas to discuss improvements in the waypoints list:
- https://github.com/devemux86/cruiser/discussions/114

**Comment by 0709wiwiwi** [comment]
App menu routing. 

- To change the routing service or change transport items is more simple and faster in the desktop version.
- Suggestion: Add the service selections in the Main App Menu > Routing. 
  By an extra (large) Button: Services.

**Reply by Leeb64** [reply]
good Idea!

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the suggestion!

In addition to the routing service,
then the routing options could also be requested to be more easily accessible in the front menu?

**Reply by Leeb64** [reply]
just jump to the menu:
Settings/ "Routing"

**Reply by devemux86** [reply]
Open directly the router settings is already available. :slightly_smiling_face: 

- https://github.com/devemux86/cruiser/discussions/174

**Comment by devemux86** [comment]
Menu currently open their sub-menu sequentially in one direction (forward).
If we press the back button, the screens  do not return back, the menu close.

1. Do you think the back button should return to the parent menu screen?

e.g. we are in the Map screen (image 2) and press back.
Should we return to the main menu (image 1) or just close the menu?

2. Do you think the menu button should open the last used menu?

e.g. press menu button > Main menu > press Routing > Import a route
Press menu button, should it open directly the last used Routing menu?
(now it always open the main menu)

Or do you prefer the current implementation?

<img src="https://github.com/devemux86/cruiser/assets/3484020/de3e2080-39f2-46af-8afc-f10e8eb512ee" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/af6b59fb-0f8d-4976-9b30-a54ddf136d1a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c16c14a2-7cad-4d5f-91cd-bf8913ef9373" width="33%"/>

**Reply by MotoUKRider** [reply]
This process occurred to me the other day, and I found it frustrating that I couldn't go "back up 1 level".

I typically press outside of the popup dialogue (top tip!) to get back to the map view, so I'm not sure if a "Return to Map" button needs to be added.

**Reply by devemux86** [reply]
Extra buttons could be added too.
e.g. a back or up button to go back one level to the parent menu.
(like in file managers)

**Reply by WolfgangZ42** [reply]
> do you prefer the current implementation?

Yes

**Reply by jclaudej** [reply]
Hi,

personnaly I prefer the current implementation 😉

**Comment by DolomitesTom** [comment]
Ad 1. Yes
Ad 2. No
Extra back button Yes

**Comment by MotoUKRider** [comment]
Having had another think about this - I'd say that you need to find a compromise between ease-of-use and being able to offer "all the menus/settings".

1. Personally, I'd like to "back-up a level" (other than the phone's back button) if I accidentally pressed the wrong button (switching between portrait+landscape makes this easy to do).
2. I'd still want to "Return to map" (pressing outside of the popup dialog permits this) but I'd say don't be afraid to use the whole screen if it allows better layout/style of menus.
3. Subscriptions, I feel, could be put into "Settings" - unless you're in trial or near the end of the subscription then you can utilize it or "nag" the user in some way.)

**Reply by 0709wiwiwi** [reply]
> 1. Personally, I'd like to "back-up a level"

Undo - Redo ?

**Comment by devemux86** [comment]
Do you have a preference for the back menu action?

We could have it in the title (for less space) or at the bottom as an icon or text:

<img src="https://github.com/devemux86/cruiser/assets/3484020/fbf63ac3-a2a6-4262-81c1-868b358a729a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/eb3c3972-7ba3-40d7-a52e-1d3614071cd6" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/a5a2d159-21ec-4658-ad69-945f1f60174d" width="33%"/>

**Reply by DolomitesTom** [reply]
first variant is nice, is the tap area wide enough to use with gloves?

**Reply by MotoUKRider** [reply]
Agree about area/size (especially if using gloves).  With the advent of these nice buttons/tiles, can a compromise button be drawn, perhaps a small square (finger-sized) with the "back" icon rather than these simple text labels?

**Reply by devemux86** [reply]
> first variant is nice, is the tap area wide enough to use with gloves?

The whole title can be tapped to go back.

**Reply by devemux86** [reply]
> can a compromise button be drawn, perhaps a small square (finger-sized) with the "back" icon rather than these simple text labels?

The middle solution is like that, an icon button at the bottom with more tap area.

**Reply by DolomitesTom** [reply]
This is an argument for using the 2nd variant

**Comment by devemux86** [comment]
I can also put the bottom button at the left:

(hoping OEMs don't override default Android alignment)

<img src="https://github.com/devemux86/cruiser/assets/3484020/b2e73273-bdba-4963-8ae4-d118bfe042c5" width="33%"/>

**Reply by DolomitesTom** [reply]
Yes good

**Comment by devemux86** [comment]
Menu back button is available in [Cruiser 3.0.22](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Please let me know how you see it.

Or if you prefer it in the title, as seen above: https://github.com/devemux86/cruiser/discussions/82#discussioncomment-6049407

**Reply by DolomitesTom** [reply]
yes very good, the arrow could be a bit thicker (just in comparison with the other menu symbols)

**Reply by MotoUKRider** [reply]
Hmmmm... The landscape view is a bit cramped (the view does scroll, but not ideal)
Samsung S20 @ 2400x1080px (other settings have no effect)

![back-button-landscape](https://github.com/devemux86/cruiser/assets/125696109/022c7154-5253-4d34-9d36-0c6481a76bf0)

**Reply by devemux86** [reply]
Did you change the sizes in "Settings | Application | **Advanced**"?

Now it is better to only change the sizes in "Settings | Application | **Appearance**".

**Reply by devemux86** [reply]
Here how it should appear with default (advanced) size settings on a FHD device:

![Screenshot_1685722126](https://github.com/devemux86/cruiser/assets/3484020/36b338e4-f122-4fc5-beaa-73f525c3592b)

**Reply by MotoUKRider** [reply]
Ah yes - "Display Size" was set to 120% so my fault!  The reasoning behind that was to get bigger "glove-friendly" on-screen map buttons.  Now I have a normal menu like yours but with smaller menu/search/zoom/location buttons - I can live with that for now :)

**Reply by devemux86** [reply]
There has to be a compromise between display size setting and available space. :slightly_smiling_face: 

Now I strongly recommend to not change anymore the sizes in "Settings | Application | **Advanced**".
As these change the whole application and can cause such side effects.

It is much more useful to only change the sizes in "Settings | Application | **Appearance**".
These only change the appearance of the on screen panels.

In any case we could move the back menu button in the title and gain some vertical space.
- See the 1st image in https://github.com/devemux86/cruiser/discussions/82#discussioncomment-6049407

**Reply by MotoUKRider** [reply]
I'm happy with the "back" button where it is.

My only gripe now is that the on-screen round buttons are tied to a setting I cannot change.  It was also useful as I started using a cradle for the phone on the bike and the edges are a little "thick", and a margin tweak isn't enough.  Perhaps I should use that stylus pen more!

**Reply by devemux86** [reply]
> I started using a cradle for the phone on the bike and the edges are a little "thick"

There are detailed margin options in "Settings | Application | Display | Screen margin".
Have you checked them?

- Related topic: https://github.com/devemux86/cruiser/discussions/24

<img src="https://user-images.githubusercontent.com/3484020/214856484-97341499-8dec-4b40-9e54-2d209189e613.jpg" width="50%"/>

**Reply by MotoUKRider** [reply]
Yes :)
> a margin tweak isn't enough

I kinda liked the buttons a bit more "Fisher Price My First Sat Nav". There was more area to press, if that makes sense.

**Reply by devemux86** [reply]
> "Fisher Price My First Sat Nav"

:laughing: 

I could increase the distance between the circle buttons.

I wonder if it's worth implementing a separate size setting for them!

**Reply by MotoUKRider** [reply]
Perhaps a slight increase.  My thinking was that, for me, I set them to auto-hide which makes their size a non-issue in normal use as they just vanish after a few seconds.  As a landscape user, I find a high margin tweak restricts the map view too much.

If the size of the on-screen buttons can be a separate size, without too much hassle or menu faff, then that would probably work.

Thanks for the work you do on this app and your catering to the more adventurous of our demands - it's very much appreciated :)

**Comment by devemux86** [comment]
Let's organize better the main menu by bringing the downloads to the front:

<img src="https://github.com/devemux86/cruiser/assets/3484020/ba09f753-c724-401d-8d4c-3ed39662fe5b" width="50%"/>

**Comment by devemux86** [comment]
It could be like this:

Or we can move the Profile to the main menu, as a small button at the bottom.
(similar to Subscriptions shown [above](https://github.com/devemux86/cruiser/discussions/82#discussioncomment-6070682))

<img src="https://github.com/devemux86/cruiser/assets/3484020/6d07a213-8a94-44d4-aa62-4582d410f16b" width="50%"/>

**Reply by DolomitesTom** [reply]
Looks also nice

**Reply by WolfgangZ42** [reply]
While "BRouter (offline)" just shows which route service I am using, I can select "Subscriptions" and then get details. 
I think "BRouter (offline)" (or any other route service) should be in the headline of the menu. And "Subscriptions" I would put in settings, maybe even under "Cruiser 3.0.xx". 
Just my 2 cents ... 😏

**Reply by devemux86** [reply]
> And "Subscriptions" I would put in settings, maybe even under "Cruiser 3.0.xx".

We can't hide subscriptions too much, Google requires users to find them easily. :slightly_smiling_face:

**Comment by devemux86** [comment]
Another implementation:

<img src="https://github.com/devemux86/cruiser/assets/3484020/74bfe1e3-ad68-46d6-b72c-47fb5d8e4860" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/9919db11-fe78-4ea9-9e86-64271f42b292" width="50%"/>

**Reply by WolfgangZ42** [reply]
The extra "button" for Downloads is better, because we download not only maps. 
But the three marked "buttons" confuse me a bit. 
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/a9c2c69c-2394-49a4-8224-30d34aebad9a"/> 
Aren't they at least partially redundant? What does "Profile" mean in the main menu? And "Options" are the options for the different route services, so could be selected by just press "BRouter (offline)" - as I said I'm a little bit confused.

**Reply by devemux86** [reply]
There is nothing final, they are just quick implementations open for discussion. :slightly_smiling_face: 

Here what each button could open:

- Profile selection
- Routing service selection
- Options of selected routing service
  - Or instead open the vehicle options, i.e. the car / motorcycle screen?

<img src="https://github.com/devemux86/cruiser/assets/3484020/82e83f90-67b5-4e9f-8951-d39831975f7a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/6f075b8e-0b74-41cb-9367-562034cdcc96" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/ee7a2c77-67de-4b57-b2b0-8cd81b4a791c" width="33%"/>
